speaker/bgp_controller.go

```diff
@@ -25,42 +26,47 @@ import (
 
 	"go.universe.tf/metallb/internal/bgp"
 	"go.universe.tf/metallb/internal/config"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/labels"
 
 	"github.com/go-kit/kit/log"
 )
 
 type peer struct {
-	cfg *config.Peer
-	bgp session
+	Cfg *config.Peer
+	BGP session
 }
 
 type bgpController struct {
-	logger     log.Logger
-	myNode     string
-	nodeLabels labels.Set
-	peers      []*peer
-	svcAds     map[string][]*bgp.Advertisement
+	logger            log.Logger
+	myNode            string
+	nodeAnnotations   labels.Set
+	nodeLabels        labels.Set
+	nodePeer          *peer
```

Comment:
I've tried the other approach (storing node peers in the same slice as the regular peers). This forces me to add a boolean `nodePeer` field to the peer struct: I must be able to identify which peer is a node peer because node peers have a special treatment. IIRC it was very tricky to get a sane reconciliation logic using this approach, therefore I've opted for separating the node peer.